bus/bus.go

```diff
@@ -46,6 +47,7 @@ type (
 		FundTransaction(cs consensus.State, txn *types.Transaction, amount types.Currency, pool []types.Transaction) ([]types.OutputID, error)
 		ReleaseInputs(txn types.Transaction)
 		SignTransaction(cs consensus.State, txn *types.Transaction, toSign []types.OutputID, cf types.CoveredFields) error
+		Split(cs consensus.State, outputs int, amount, feePerByte types.Currency, pool []types.Transaction) (types.Transaction, []types.OutputID, error)
```

Comment:
In `walletd` we decided to have the handler call `wallet.DistributeFunds` instead of adding a method to the `Wallet` interface

Reply:
I removed it from the interface and copied the behaviour we have in `walletd`.

I personally preferred adding it to the interface in the case of `renterd` because it keeps the handler clean and that way we would also be able to update the `used` field in the wallet keeping track of used outputs. I wanted to explicitly mention that because currently the `split` logic does not update that field, do we care about that?

Reply:
hmm, true. `used` is important to prevent accidental UTXO reuse when multiple transactions are being constructed concurrently.

Reply:
we could revert to the commit prior to the last one and have it on the interface in `renterd` - it's both cleaner and more correct

Reply:
perhaps we could reuse `FundTransaction` instead? Having a separate `Split` method is *slightly* better because you can filter out UTXOs that already match the desired size -- but is there any other advantage? I'm inclined to say it's not worth it.

Reply:
There's another reason though unfortunately: `FundTransaction` uses random UTXO ordering which is not ideal for split transactions, it's actually the reason why I did not do it in the first place. I considered adding an extra parameter to `FundTransaction` to signal what type of ordering you want but I felt at that point I was better off adding an extra `Split` method instead.

Reply:
ah, right. Ok then